platforms/ios/src/TangramMap/TGMapViewController.mm

```diff
@@ -535,56 +535,135 @@ - (void)setCameraType:(TGCameraType)cameraType
     // Distanle single tap when double tap occurs
-    [tapRecognizer requireGestureRecognizerToFail:doubleTapRecognizer];
+    [_tapGestureRecognizer requireGestureRecognizerToFail:_doubleTapGestureRecognizer];
 
-    //3. Pan
-    UIPanGestureRecognizer* panRecognizer = [[UIPanGestureRecognizer alloc]
-                                             initWithTarget:self action:@selector(respondToPanGesture:)];
-    panRecognizer.maximumNumberOfTouches = 1;
+    _panGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(respondToPanGesture:)];
+    _panGestureRecognizer.maximumNumberOfTouches = 1;
 
-    //4. Pinch
-    UIPinchGestureRecognizer* pinchRecognizer = [[UIPinchGestureRecognizer alloc]
-                                                 initWithTarget:self action:@selector(respondToPinchGesture:)];
+    _pinchGestureRecognizer = [[UIPinchGestureRecognizer alloc] initWithTarget:self action:@selector(respondToPinchGesture:)];
+    _rotationGestureRecognizer = [[UIRotationGestureRecognizer alloc] initWithTarget:self action:@selector(respondToRotationGesture:)];
+    _shoveGestureRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(respondToShoveGesture:)];
+    _shoveGestureRecognizer.minimumNumberOfTouches = 2;
+    _longPressGestureRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(respondToLongPressGesture:)];
 
-    //5. Rotate
-    UIRotationGestureRecognizer* rotationRecognizer = [[UIRotationGestureRecognizer alloc]
-                                                       initWithTarget:self action:@selector(respondToRotationGesture:)];
+    // Use the delegate method 'shouldRecognizeSimultaneouslyWithGestureRecognizer' for gestures that can be concurrent
+    _panGestureRecognizer.delegate = self;
+    _pinchGestureRecognizer.delegate = self;
+    _rotationGestureRecognizer.delegate = self;
 
-    //6. Shove
-    UIPanGestureRecognizer* shoveRecognizer = [[UIPanGestureRecognizer alloc]
-                                               initWithTarget:self action:@selector(respondToShoveGesture:)];
-    shoveRecognizer.minimumNumberOfTouches = 2;
+    [self.view addGestureRecognizer:_tapGestureRecognizer];
+    [self.view addGestureRecognizer:_doubleTapGestureRecognizer];
+    [self.view addGestureRecognizer:_panGestureRecognizer];
+    [self.view addGestureRecognizer:_pinchGestureRecognizer];
+    [self.view addGestureRecognizer:_rotationGestureRecognizer];
+    [self.view addGestureRecognizer:_shoveGestureRecognizer];
+    [self.view addGestureRecognizer:_longPressGestureRecognizer];
+}
 
-    //7. Long press
-    UILongPressGestureRecognizer* longPressRecognizer = [[UILongPressGestureRecognizer alloc]
-                                                         initWithTarget:self action:@selector(respondToLongPressGesture:)];
+- (void) setTapGestureRecognizer:(UITapGestureRecognizer *)recognizer
```

Comment:
Minor nitpick: I think the rest of the file is styled like
`- (void)setTapGestureRecognizer`
whereas you have a space after the `)` .